catalogs/catalog-lakehouse-hudi/src/main/java/org/apache/gravitino/catalog/hudi/HudiCatalogOperations.java

```diff
@@ -0,0 +1,258 @@
+import org.apache.gravitino.catalog.hudi.ops.HudiCatalogOps;
+import org.apache.gravitino.catalog.hudi.utils.CatalogUtils;
+import org.apache.gravitino.connector.CatalogInfo;
+import org.apache.gravitino.connector.CatalogOperations;
+import org.apache.gravitino.connector.HasPropertyMetadata;
+import org.apache.gravitino.connector.SupportsSchemas;
+import org.apache.gravitino.exceptions.ConnectionFailedException;
+import org.apache.gravitino.exceptions.NoSuchCatalogException;
+import org.apache.gravitino.exceptions.NoSuchSchemaException;
+import org.apache.gravitino.exceptions.NoSuchTableException;
+import org.apache.gravitino.exceptions.NonEmptySchemaException;
+import org.apache.gravitino.exceptions.SchemaAlreadyExistsException;
+import org.apache.gravitino.exceptions.TableAlreadyExistsException;
+import org.apache.gravitino.rel.Column;
+import org.apache.gravitino.rel.Table;
+import org.apache.gravitino.rel.TableCatalog;
+import org.apache.gravitino.rel.TableChange;
+import org.apache.gravitino.rel.expressions.distributions.Distribution;
+import org.apache.gravitino.rel.expressions.sorts.SortOrder;
+import org.apache.gravitino.rel.expressions.transforms.Transform;
+import org.apache.gravitino.rel.indexes.Index;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Operations for Interacting with Hudi Catalog. */
+public class HudiCatalogOperations implements CatalogOperations, SupportsSchemas, TableCatalog {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HudiCatalogOperations.class);
+
+  private HudiCatalogOps hudiCatalogOps;
+
+  /**
+   * Load the Hudi Catalog Backend and initialize the Hudi Catalog Operations.
+   *
+   * @param config The configuration of this Catalog.
+   * @param info The information of this Catalog.
+   * @param propertiesMetadata The properties metadata of this Catalog.
+   * @throws RuntimeException if failed to initialize the Hudi Catalog Operations.
+   */
+  @Override
+  public void initialize(
+      Map<String, String> config, CatalogInfo info, HasPropertyMetadata propertiesMetadata)
+      throws RuntimeException {
+    HudiCatalogBackend hudiCatalogBackend = CatalogUtils.loadHudiCatalogBackend(config);
+    hudiCatalogOps = hudiCatalogBackend.catalogOps();
+  }
+
+  /**
+   * Performs `listSchemas` operation on the Hudi Catalog to test the catalog connection.
+   *
+   * @param catalogIdent the name of the catalog.
+   * @param type the type of the catalog.
+   * @param provider the provider of the catalog.
+   * @param comment the comment of the catalog.
+   * @param properties the properties of the catalog.
+   * @throws Exception if failed to run `listSchemas` operation on the Hudi Catalog.
+   */
+  @Override
+  public void testConnection(
+      NameIdentifier catalogIdent,
+      Catalog.Type type,
+      String provider,
+      String comment,
+      Map<String, String> properties)
+      throws Exception {
+    try {
+      hudiCatalogOps.listSchemas(null);
```

Comment:
Is there other solution to implement the `testConnection` for example whether the  `hudiCatalogOps` is successfully initialized?